app/connectors/exchange/ExportsDeclarationExchange.scala

```diff
@@ -68,9 +73,10 @@ case class ExportsDeclarationExchange(
     locations = this.locations,
     items = this.items,
     totalNumberOfItems = this.totalNumberOfItems.flatMap { exchange =>
-      (exchange.totalAmountInvoiced, exchange.exchangeRate) match {
-        case (None, None)                        => None
-        case (totalAmountInvoiced, exchangeRate) => Some(TotalNumberOfItems(exchangeRate, totalAmountInvoiced))
+      (exchange.totalAmountInvoiced, exchange.totalAmountInvoicedCurrency, exchange.exchangeRate) match {
```

Comment:
Probably I miss something, but why do you need pattern matching here. Can't you directly be doing something like...
```
Some(TotalNumberOfItems(
    exchange.exchangeRate,
    exchange.totalAmountInvoiced,
    exchange.totalAmountInvoicedCurrency
))
```

Reply:
Good question. On closer inspection it appears I've broken the original intent of the code to not populate the field with a `Some(TotalNumberOfItems(None, None, None))`. I'll correct this :+1: